src/provider.ts

```diff
@@ -17,44 +18,64 @@ import {
  * Provided by marina extension at window.marina
  */
 export interface MarinaProvider {
+  // ask the extension for authorization (by hostname)
   enable(): Promise<void>;
-
+  // disable access to the extension from the current hostname
   disable(): Promise<void>;
-
+  // return true if the current hostname calling this method is authorized to access marina's provider instance
   isEnabled(): Promise<boolean>;
+  // return true if the current marina extension is ready to use connected apps
+  // i.e user done the onboarding process
+  isReady(): Promise<boolean>;
 
-  setAccount(account: number): Promise<void>;
+  // set up a listener for an event type
+  on(type: MarinaEventType, callback: (payload: any) => void): EventListenerID;
+  // remove a listener for an event type
+  off(listenerId: EventListenerID): void;
 
+  // the current network the extension is connected to
   getNetwork(): Promise<NetworkString>;
+  // return the assets accepted as network fees
+  getFeeAssets(): Promise<string[]>;
+  // which account is currently selected
+  getSelectedAccount(): Promise<string>;
 
-  getAddresses(): Promise<AddressInterface[]>;
+  createAccount(accountName: string): Promise<void>;
 
-  getNextAddress(): Promise<AddressInterface>;
+  // select an account
+  // return true if the account is ready to be used,
+  // false if u need to set up the script templates
+  useAccount(account: string): Promise<boolean>;
 
+  /** all the methods above apply to the selected account **/
+
+  // set up descriptor templates for the current account
+  // fails if the account has already a template
+  // if not setup, changeTemplate = template
+  importTemplate(
```

Comment:
how do we let user signals to us that one leaf in a TREE of a  `eltr` template needsParamaters?

Reply:
(assuming we automatically detect for each leaf `hasIntrospection` and `needsSignature`)

Reply:
Shalle we ask user to fill `needParameters` on each template?